File: DemoApp Objective-C/CASSample/SupportFiles/Main/Enums/FormatsSection.m
//
//  FormatsSection.m
//  CASSample
//

#import "FormatsSection.h"

@implementation FormatsSectionHelper

+ (NSArray<NSNumber *> *)allCases {
    return @[
        @(FormatsSectionAppOpen),
        @(FormatsSectionBanner),
        @(FormatsSectionNative),
        @(FormatsSectionNativeTemplate),
        @(FormatsSectionInterstitial),
        @(FormatsSectionRewarded)
    ];
}

+ (NSString *)titleForFormat:(FormatsSection)format {
    switch (format) {
        case FormatsSectionAppOpen:
            return @"App Open";
        case FormatsSectionBanner:
            return @"Banner";
        case FormatsSectionNative:
            return @"Native";
        case FormatsSectionNativeTemplate:
            return @"Native Template";
        case FormatsSectionInterstitial:
            return @"Interstitial";
        case FormatsSectionRewarded:
            return @"Rewarded";
    }
}

@end

File: DemoApp Swift/CASSample/SupportFiles/Main/Enums/FormatsSection.swift
//
//  FormatsSection.swift
//  CASSample
//
//  Copyright © 2025 Clever Ads Solutions. All rights reserved.
//

import Foundation

enum FormatsSection: CaseIterable {
    case appOpen
    case banner
    case native
    case nativeTemplate
    case interstitial
    case rewarded
    
    var title: String {
        switch self {
        case .appOpen:
            return "App Open"
        case .banner:
            return "Banner"
        case .native:
            return "Native"
        case .nativeTemplate:
            return "Native Template"
        case .interstitial:
            return "Interstitial"
        case .rewarded:
            return "Rewarded"
        }
    }
}
